dali/base/dadfs.cpp

```diff
@@ -586,6 +587,56 @@ class CFileSuperOwnerLock : protected CFileLockCompound
     {
         return CFileLockCompound::detach();
     }
+    bool initWithFileLock(const CDfsLogicalFileName &logicalName, IRemoteConnection *conn, unsigned timeout, const char *msg, CFileLock &fcl, unsigned fclmode)
+    {
+        // SuperOwnerLock while holding fcl
+        IRemoteConnection *fclConn = fcl.queryConnection();
+        if (!fclConn)
+            return false; // throw ?
+        CTimeMon tm(timeout);
+        unsigned remaining = timeout;
+        unsigned interval = 5000;
+        loop
+        {
+            try
+            {
+                if (init(logicalName, conn, 0, msg))
+                    return true;
+                else
+                    return false; // throw ?
+            }
+            catch (ISDSException *e)
+            {
+                if (SDSExcpt_LockTimeout != e->errorCode() || tm.timedout(&remaining))
+                    throw;
+                e->Release();
+            }
+            unsigned delta = remaining;
+            if (delta < interval)
+                delta = interval;
+            // release lock
+            {
+                CriticalBlock block(superOwnerLockSect);
```

Comment:
Am not sure why the crit was needed? 
There can't be any other threads dealing with fcl in contexts used and would likely need more changes if fcl object was intended to be thread-safe.